File: lib-segment/src/main/java/websiteschema/mpsegment/util/ByteArrayReader.java
package websiteschema.mpsegment.util;

import java.io.DataInputStream;
import java.io.IOException;
import java.io.InputStream;

public class ByteArrayReader
        implements BufReader {

    DataInputStream input = null;

    public ByteArrayReader(InputStream stream) {
        input = new DataInputStream(stream);
    }

    @Override
    public void close()
            throws IOException {
        input.close();
        input = null;
    }

    @Override
    public int read()
            throws IOException {
        return input.readByte();
    }

    @Override
    public byte readByte()
            throws IOException {
        return input.readByte();
    }

    @Override
    public int readIntByte()
            throws IOException {
        return read();
    }

    @Override
    public int readInt()
            throws IOException {
        return input.readInt();
    }

    @Override
    public int read(byte abyte0[])
            throws IOException {
        return input.read(abyte0);
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/util/ByteArrayReader.scala
package websiteschema.mpsegment.util;

import java.io.{InputStream, DataInputStream}

class ByteArrayReader(stream:InputStream)
        extends BufReader {

    val input = new DataInputStream(stream);

    override def close(): Unit = input.close()

    override def read(): Int = input.readByte()

    override def readByte():Byte = input.readByte()

    override def readIntByte(): Int = read()

    override def readInt(): Int =  input.readInt()

    override def read(bytes: Array[Byte]): Int = input.read(bytes)
}
